### PR TITLE
chore: [CDS-80993]: update UX for runtime input drawer

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

- Moved search bar from header to content
- Add new input flow is inline in split pane rather than modal

https://github.com/harness/harness-core-ui/assets/100121280/3437cd0f-bb19-4302-8fea-f707f5d2a840

Test scenarios:
- New feature for yaml_simplification 

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by srdjan-harness — APPROVED** [review]

File: src/modules/70-pipeline/y1/components/RuntimeInputs/RuntimeInputList.tsx [hunk 1]
```diff
@@ -50,20 +51,26 @@ interface ValidateOptionsExtended {
...
 interface RuntimeInputsListWithRefProps {
-  searchTerm: string
   formikRef: React.MutableRefObject<FormikProps<RuntimeInputsFormData> | undefined>
+  pipeline: PipelineInfoConfig
+  isReadonly: boolean
+}
+
+const SEARCH_DEBOUNCE_DELAY = 300
+
+export enum RuntimeInputDataTabs {
```
Comment (rameshx) [inline src/modules/70-pipeline/y1/components/RuntimeInputs/RuntimeInputList.tsx:61]:
```suggestion
export enum RuntimeInputDataTab {
```

File: src/modules/70-pipeline/y1/components/RightDrawer/RightDrawerY1.tsx [hunk 2]
```diff
@@ -20,8 +32,18 @@ export function RightDrawerY1(): JSX.Element {
     updatePipelineView({ ...pipelineView, isDrawerOpened: false, drawerData: { type: DrawerTypesY1.AddStep } })
   }
 
+  const onUpdateInputs = async (updatedInputs: PipelineInputs): Promise<void> => {
+    const updatedPipeline = produce(pipeline, draft => {
+      set(draft, 'inputs', updatedInputs)
```
Comment (rameshx) [inline src/modules/70-pipeline/y1/components/RightDrawer/RightDrawerY1.tsx:37]: `inputs` will be under `spec` in the updated schema. Let's update the path here and at the place where the initial inputs are read.

**Comment by rameshx** [comment]
The search input's width should be reduced to prevent the Add Runtime Input button overflow.

image

**Comment by PratyushGar** [comment]
> The search input's width should be reduced to prevent the Add Runtime Input button overflow.
> 
> image

Sure, addressed
image

**Review by rameshx — APPROVED** [review]

**Pull request merged**